src/ragas/experimental/testset/transforms/extractors/regex_based.py

```diff
@@ -0,0 +1,43 @@
+import re
+import typing as t
+from dataclasses import dataclass
+
+from ragas.experimental.testset.graph import Node
+from ragas.experimental.testset.transforms.base import Extractor
+
+
+@dataclass
+class RegexBasedExtractor(Extractor):
+    pattern: str = ""
+    is_multiline: bool = False
+    property_name: str = "regex"
+
+    async def extract(self, node: Node) -> t.Tuple[str, t.Any]:
+        text = node.get_property("page_content")
+        if not isinstance(text, str):
+            raise ValueError(
+                f"node.property('page_content') must be a string, found '{type(text)}'"
+            )
+
+        matches = re.findall(self.pattern, text, re.MULTILINE)
+        return self.property_name, matches
+
+
+# This regex pattern matches URLs, including those starting with "http://", "https://", or "www."
+links_extractor_pattern = r"(?i)\b(?:https?://|www\.)\S+\b"
```

Comment:
here we are initing these extractors - should we change that?

Reply:
Yes, please. Such inits can't be scaled cleanly.